### PR TITLE
CLN FloatArrayFormatter

### PR DESCRIPTION
**Pull request by nbonnotte**
I'm working on #12164

This is a work in progress.

### CODE REVIEW

**Comment by nbonnotte** [comment]
Ok, that should do.

What I have done:

- Instead of differentiating on the one hand `FloatArrayFormatter.get_result` (which comes from `GenericArrayFormatter`), and on the other hand `FloatArrayFormatter.get_formated_data`, according to their destination (two usages, two functions, two independent pieces of code), I differentiate with the type of the output: `.get_result` keeps returning a list like before, and the new `.get_result_as_array` returns a numpy array like `.get_formatted_data` used to.

- The question "do we want a fixed-width output?", which is a formatting question, is translated like all formatting matters into a parameter `fixed_width` for the `GenericArrayFormatter` constructor (boolean, default `True`, because that's the main usage as of today)

This seems (to me) the simplest way to know which method should be used.

I have merged the two independent pieces of code that we used to have, and which were doing basically the same thing. But `FloatArrayFormatter.get_formated_data` was more refined, as it was using a mask to deal separately with the NaN values. I've kept that approach, but I don't know if it is the best option in term of performances.

File: pandas/core/format.py [hunk 1]
```diff
@@ -2076,94 +2077,136 @@ class FloatArrayFormatter(GenericArrayFormatter):
...
+            threshold = None
+
+        # if we have a fixed_width, we'll need to try different float_format
+        def format_values_with(float_format):
+            formatter = self._value_formatter(float_format, threshold)
+
+            # separate the wheat from the chaff
+            values = self.values
+            mask = isnull(values)
+            if hasattr(values, 'to_dense'):  # sparse array
```
Comment (jreback) [inline pandas/core/format.py]: you can always call this, ``to_dense()`` is a generic method (doesn't do anything to non-dense things)

Comment (nbonnotte) [inline pandas/core/format.py]: `AttributeError: 'numpy.ndarray' object has no attribute 'to_dense'`

:)

**Comment by nbonnotte** [comment]
All green

**Comment by jreback** [comment]
this looked ok last time. pls rebase and repush.

**Comment by nbonnotte** [comment]
@jreback all set

**Comment by jreback** [comment]
does this actually change any features? e.g. implement the decimal fix? (or is that another PR), this just cleanup?

**Comment by jreback** [comment]
thanks!

**Comment by nbonnotte** [comment]
No, that was just a cleanup, that's why I haven't touched the tests. Now I should be able to move on to the decimal parameter (issue #12031)

**Pull request closed without merging**